scripts/fetch-esgf.py

```diff
@@ -0,0 +1,360 @@
+This fetches about 3TB of datasets into the default location for intake esgf.
+This can be adjusted via `~/.config/intake-esgf/config.yaml`.
+"""
+
+import intake_esgf
+import typer
+from attrs import define
+from loguru import logger
+
+
+@define
+class CMIP6Request:
+    """
+    A set of CMIP6 data that will be fetched from ESGF
+    """
+
+    id: str
+    facets: dict[str, str | tuple[str, ...] | list[str]]
+
+    def fetch(self, remove_ensembles: bool = True):
+        """
+        Fetch CMIP6 data from the ESGF catalog and return it as a DataFrame.
+
+        Parameters
+        ----------
+        remove_ensembles : bool, default True
+            Whether to remove ensemble members, keeping only one per model.
+            If False, all ensemble members will be included.
+
+        Returns
+        -------
+        pd.DataFrame
+            DataFrame containing the metadata for the CMIP6 datasets.
+        """
+        catalog = intake_esgf.ESGFCatalog()
+        search_parameters = {
+            "project": "CMIP6",
+            "frequency": ["mon", "fx"],
+            **self.facets,
+        }
+        logger.debug(f"Fetching CMIP6 data: {search_parameters}")
+        try:
+            cmip6_data = catalog.search(**search_parameters)
+            if remove_ensembles:
+                cmip6_data = cmip6_data.remove_ensembles()
+            return cmip6_data.to_path_dict()
+        except Exception:
+            logger.info(f"Error fetching CMIP6 data: {search_parameters}")
+        return {}
+
+
+@define
+class Obs4MIPsRequest:
+    """
+    A set of Obs4MIPs data that will be fetched from ESGF
+    """
+
+    id: str
+    facets: dict[str, str | tuple[str, ...] | list[str]]
+
+    def fetch(self, remove_ensembles: bool = True):
+        """
+        Fetch Obs4MIPs data from the ESGF catalog and return it as a DataFrame.
+
+        Parameters
+        ----------
+        remove_ensembles : bool, default True
+            Ignored as Obs4MIPs data does not have ensembles.
+
+        Returns
+        -------
+        pd.DataFrame
+            DataFrame containing the metadata for the Obs4MIPs datasets.
+        """
+        catalog = intake_esgf.ESGFCatalog()
+        search_parameters = {
+            "project": "obs4MIPs",
+            "frequency": ["mon", "fx"],
+            **self.facets,
+        }
+        logger.info(f"Fetching Obs4MIPs data: {search_parameters}")
+        try:
+            obs_data = catalog.search(**search_parameters)
+            return obs_data.to_path_dict(minimal_keys=False)
+        except Exception:
+            logger.exception(f"Error fetching Obs4MIPs data: {search_parameters}")
+        return {}
+
+
+Request = CMIP6Request | Obs4MIPsRequest
+
+requests: list[Request] = [
+    CMIP6Request(
+        id="esmvaltool-climate-at-global-warmings-levels",
+        facets=dict(
+            variable_id=["pr", "tas"],
+            experiment_id=["ssp126", "ssp245", "ssp370", "ssp585", "historical"],
+        ),
```

Comment:
You could reduce the data volume by specifying the `table_id` according to how the ESMValTool diagnostics do it, e.g. https://github.com/Climate-REF/climate-ref/blob/27b28bffd4d8ba13b294f3f0d003102f348a9e44/packages/climate-ref-esmvaltool/src/climate_ref_esmvaltool/diagnostics/cloud_radiative_effects.py#L41 and refined in #414

Reply:
Added the extra table_id's. 

It would have been cool to explicitly link the data requirements, but doing it manually was likely quicker at this point in time